code/modules/power/cell.dm

```diff
@@ -148,6 +148,30 @@
 				if(prob(25))
 					corrupt()
 
+/obj/item/stock_parts/cell/attack_self(mob/user)
+	var/mob/living/carbon/human/H = user
+	if(istype(H))
+		var/datum/species/ethereal/eth_species = H.dna?.species
+		if(istype(eth_species))
+			if(charge < 25)
+				to_chat(H, "<span class='warning'>The [src] doesn't have enough power!</span>")
+				return
+			var/obj/item/organ/stomach/ethereal/stomach = H.getorganslot(ORGAN_SLOT_STOMACH)
+			if(stomach.crystal_charge > 96)
+				to_chat(H, "<span class='warning'>Your charge is full!</span>")
+				return
+			to_chat(H, "<span class='notice'>You clumsily channel power through the [src] and into your body, wasting some in the process.</span>")
+			if(do_after(user, 5, target = src))
+				if((charge < 25) || (stomach.crystal_charge > 96))
+					return
+				if(istype(stomach))
+					to_chat(H, "<span class='notice'>You receive some charge from the [src].</span>")
+					stomach.adjust_charge(3)
+					charge -= 100 //you waste way more than you receive, so that ethereals cant just steal one cell and forget about hunger

```

Comment:
this exchange rate is a bit TOO awful, imo.

maybe make it, like, 3 gained (by you):30 lost (by the cell)?

also, what happens if the cell has less than 100 charge but more than 25?

Reply:
that 25 is an old value, I’ll need to change it. 

also the gain vs lost ratio is fine to me. If anything ethereals still get too much

Reply:
still need to fix the 25 here